### PR TITLE
Add federator.pegout.enabled optional setting

### PR DESCRIPTION
**Pull request by marcos-iov**

### CODE REVIEW

File: src/main/java/co/rsk/federate/config/FedNodeSystemProperties.java [hunk 1]
```diff
@@ -20,6 +20,10 @@ public boolean isFederatorEnabled() {
         return configFromFiles.hasPath("federator.enabled") && configFromFiles.getBoolean("federator.enabled");
     }
 
+    public boolean isPegoutEnabled() {
+        return !configFromFiles.hasPath("federator.pegout.enabled") || configFromFiles.getBoolean("federator.pegout.enabled");
```
Comment (kelvinator07) [inline src/main/java/co/rsk/federate/config/FedNodeSystemProperties.java:24]:
```suggestion
        return configFromFiles.hasPath("federator.pegout.enabled") && configFromFiles.getBoolean("federator.pegout.enabled");
```
Isnt This Simpler To Read?

Comment (nathanieliov) [reply src/main/java/co/rsk/federate/config/FedNodeSystemProperties.java:24]: Nice K! I like yours. Looks simpler to read for me too.

**Review by nathanieliov — APPROVED** [review]
LGTM

File: src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientTest.java [hunk 2]
```diff
@@ -407,26 +416,73 @@ public void onBestBlock_return_when_node_is_syncing() throws BtcReleaseClientExc
...
+            mock(BtcReleaseClientStorageAccessor.class),
+            mock(BtcReleaseClientStorageSynchronizer.class)
+        );
+        btcReleaseClient.start(federation);
+
+        // Act
+        ethereumListener.get().onBestBlock(null, null);
+
+        // Assert
+        verify(federatorSupport, never()).getStateForFederator();
```
Comment (nathanieliov) [inline src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientTest.java:465]: Let's add one more assertion

```suggestion
        verify(nodeBlockProcessor, times(1)).hasBetterBlockToSync();
        verify(federatorSupport, never()).getStateForFederator();
```

File: src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientTest.java [hunk 3]
```diff
@@ -439,16 +495,56 @@ public void onBlock_return_when_node_is_syncing() {
...
 
+        List<TransactionReceipt> receipts = new ArrayList<>();
+        TransactionReceipt transactionReceipt = mock(TransactionReceipt.class);
+        receipts.add(transactionReceipt);
+
+        // Act
+        ethereumListener.get().onBlock(null, receipts);
+
+        // Assert
+        verifyZeroInteractions(transactionReceipt);
```
Comment (nathanieliov) [inline src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientTest.java:506]: Let's add one more assertion:

```suggestion
        verify(nodeBlockProcessor, times(1)).hasBetterBlockToSync();
        verifyZeroInteractions(transactionReceipt);
```

File: src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientTest.java [hunk 4]
```diff
@@ -439,16 +495,56 @@ public void onBlock_return_when_node_is_syncing() {
...
-        receipts.add(mocktTransactionReceipt);
+        TransactionReceipt transactionReceipt = mock(TransactionReceipt.class);
+        receipts.add(transactionReceipt);
 
         // Act
         ethereumListener.get().onBlock(null, receipts);
 
         // Assert
-        Mockito.verifyZeroInteractions(mocktTransactionReceipt);
+        verifyZeroInteractions(transactionReceipt);
```
Comment (nathanieliov) [inline src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientTest.java:547]: And here,

Let's add also one more assertion.

```suggestion
        verify(nodeBlockProcessor, never()).hasBetterBlockToSync();
        verifyZeroInteractions(transactionReceipt);
```

**Review by nathanieliov — COMMENTED** [review]
Well done Marcos. Just two minor suggestions.

**Pull request merged**